### ISSUE TITLE
Provide possibility to commit and abort transactions just by providing the transaction id (and not the struct)

### ISSUE THREAD

**Issue by Kafkalasch** [arangodb/go-driver#646]
To commit or abort a transaction, only the the transactionId is needed. The interface of the old arangodb driver (v1.x) did reflect this and allowed a simple usage.

The new interface does not allow this any more. I cannot just store the transactionId in my system to be able to commit or abort a transaction, I need to actually store the `arangodb.Transaction` struct somewhere which leads to internal non-serializable structs bleeding into my code and making everything messier.

Would it be possible, to either provide `Commit(context, transactionId, opts...)` and `Abort(context, transactionid, opts)` methods on the database level?
If not, then at least expose a public method to create a Transaction struct from the db and transactionId?

Right now, the only workaround is to create a new Transaction object by calling `Transaction(..)` on the db, but this always triggers a complete unnecessary http requests.

**Comment by Kafkalasch** [comment]
Ah, I see. This won't work, because you have removed the possibility to provide the transactionId via the context object.

Oh, what a pity. I really liked how this was done, as you could have one code basis and simply by providing a transactionId via the context decide if the methods were executed as part of a transaction or not.

Basically exactly like the API works and very clean and decoupled.

**Comment by Kafkalasch** [comment]
If anyone comes across this:

I basically reimplemented the old behaviour by doing this:

- First I created a wrapper around the `connection.Connection` object. This wrapper simply passes all the interface methods. Only for `Do` and `Stream` it first checks the `ctx` if it contains a transactionId. If so, it will modify the request accordingly and pass it down to the wrapped connection.
- Second. I created my own `CommitTransaction` and `AbortTransaction` methods. In the end, it is just calling an endpoint.
- My `BeginTransaction` endpoint does the same as the arangodb driver, but instead of returning a `Transaction` object, it returns a `Context` object, which I pass on through the following code.

This way, no arangodb driver specific structs bleed into my code and I can hide them away behind some generic interface.
Also, I have the same behaviour as in arango driver version 1.x and my migration is easier.
